test/unit/sdam/server_description.test.ts

```diff
@@ -0,0 +1,156 @@
+import { expect } from 'chai';
+
+import { Long, ObjectId } from '../../../src/bson';
+import {
+  compareTopologyVersion,
+  ServerDescription,
+  TopologyVersion
+} from '../../../src/sdam/server_description';
+
+describe('ServerDescription', function () {
+  describe('error equality', function () {
+    const errorEqualityTests = [
+      {
+        description: 'equal error types and messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: true
+      },
+      {
+        description: 'equal error types and unequal messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('blah') }),
+        equal: false
+      },
+      {
+        description: 'unequal error types and equal messages',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new TypeError('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: false
+      },
+      {
+        description: 'null lhs',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: null }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new Error('test') }),
+        equal: false
+      },
+      {
+        description: 'null rhs',
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new TypeError('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: undefined }),
+        equal: false
+      }
+    ];
+
+    for (const test of errorEqualityTests) {
+      it(test.description, function () {
+        expect(test.lhs.equals(test.rhs)).to.equal(test.equal);
+      });
+    }
+  });
+
+  it('should sensibly parse an ipv6 address', function () {
+    const description = new ServerDescription('[ABCD:f::abcd:abcd:abcd:abcd]:27017');
+    expect(description.host).to.equal('abcd:f::abcd:abcd:abcd:abcd');
+    expect(description.port).to.equal(27017);
+  });
+
+  describe('compareTopologyVersion()', () => {
+    const processIdZero = new ObjectId('00'.repeat(12));
+    const processIdOne = new ObjectId('00'.repeat(11) + '01');
+    type CompareTopologyVersionTest = {
+      title: string;
+      currentTv?: TopologyVersion;
+      newTv?: TopologyVersion;
+      out: 0 | -1 | 1;
+    };
+
+    const compareTopologyVersionEqualTests: CompareTopologyVersionTest[] = [
```

Comment:
Would it be worth also having non-zero Longs in the counter, for completeness?

Reply:
Yep, added!